envelope.js

```diff
@@ -68,13 +69,30 @@ module.exports = function Envelope (keystore, state) {
       readKey = unboxKey(envelope, feed_id, prev_msg_id, trial_own_keys, { maxAttempts: 16 })
       if (readKey) return readKey
     } else {
-      const trial_dm_keys = keystore.decryptionKeys(author).dm
+      // const trial_dm_keys = [keystore.dm.get(state.keys.id, author)]
+      /// ///
+      function addDMPairSync (myKeys, theirId) {
+        // if (!keyringReady.ready) throw new Error('keyring not ready')
+        const myId = myKeys.id
+        const myDhKeys = new DHKeys(myKeys, { fromEd25519: true })
+        const theirKeys = { public: bfe.encode(theirId).slice(2) }
+        const theirDhKeys = new DHKeys(theirKeys, { fromEd25519: true })
+        return keystore.dm.add(myId, theirId, myDhKeys, theirDhKeys, (err) => {
+          if (err) console.error(err)
+        })
+      }
+
+      if (!keystore.dm.has(state.keys.id, author)) addDMPairSync(state.keys, author)
+      // const dmKey = keyring.dm.get(authorKeys.id, recp)
+      const trial_dm_keys = [keystore.dm.get(state.keys.id, author)]
+
+      /// ///
```

Comment:
I was getting undefined keys here when simply trying to `dm.get` them, so stole this chunk of code from `ssb-box2/format.js` and it seems to have improved things?